hazelcast/src/main/java/com/hazelcast/query/impl/BaseIndexStore.java

```diff
@@ -227,6 +243,17 @@ private Comparable sanitizeScalar(Object input) {
         }
     }
 
+    void markIndexStoreExpirableIfNecessary(QueryableEntry record) {
+        assert lock.isWriteLockedByCurrentThread();
+        if (record.getStoreAdapter() != null) {
```

Comment:
Is this null check here because of query cache indexes?

Reply:
We also have plenty of unit tests where we test indexes in isolated way and even don't have record store.

Reply:
I see, could you please add a clarifying comment to the source code here and for the `record != null` check in `QueryableEntry`.